packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1369,6 +1377,73 @@ function dispatchAction<S, A>(
   }
 }
 
+const noOpMount = () => {};
+
+function mountEventListener(event: ReactListenerEvent): ReactListenerMap {
+  if (enableUseEventAPI) {
+    const hook = mountWorkInProgressHook();
+
+    const clear = () => {
+      // TODO
+    };
+
+    const reactListenerMap: ReactListenerMap = {
+      clear,
+      setListener(instance: EventTarget, callback: ?(Event) => void): void {
+        // TODO
+      },
+    };
+    // In order to clear up upon the hook unmounting,
+    /// we ensure we push an effect that handles the use-case.
+    currentlyRenderingFiber.effectTag |= UpdateEffect;
+    pushEffect(NoHookEffect, noOpMount, clear, null);
+    hook.memoizedState = [reactListenerMap, event, clear];
+    return reactListenerMap;
+  }
+  // To make Flow not complain
+  return (undefined: any);
+}
+
+function updateEventListener(event: ReactListenerEvent): ReactListenerMap {
+  if (enableUseEventAPI) {
+    const hook = updateWorkInProgressHook();
+    const [reactListenerMap, memoizedEvent, clear] = hook.memoizedState;
+    if (__DEV__) {
+      if (memoizedEvent.type !== event.type) {
+        console.warn(
+          'The event type argument passed to the useEvent() hook was different between renders.' +
+            ' The event type is static and should never change between renders.',
+        );
+      }
+      if (memoizedEvent.capture !== event.capture) {
+        console.warn(
+          'The "capture" option passed to the useEvent() hook was different between renders.' +
+            ' The "capture" option is static and should never change between renders.',
+        );
+      }
+      if (memoizedEvent.priority !== event.priority) {
+        console.warn(
+          'The "priority" option passed to the useEvent() hook was different between renders.' +
+            ' The "priority" option is static and should never change between renders.',
+        );
+      }
+      if (memoizedEvent.passive !== event.passive) {
+        console.warn(
+          'The "passive" option passed to the useEvent() hook was different between renders.' +
+            ' The "passive" option is static and should never change between renders.',
+        );
+      }
+    }
+    // In order to clear up upon the hook unmounting,
+    /// we ensure we push an effect that handles the use-case.
```

Comment:
> In order to clear up upon the hook unmounting,

What does this mean? Not sure I understand.

nit
```suggestion
    // we ensure we push an effect that handles the use-case.
```

Reply:
I updated the comment. Hopefully it makes more sense now!